Code/Conviva/Source/Events/ConvivaReporters/AppEventConvivaReporter.swift

```diff
@@ -4,13 +4,38 @@
 
 import ConvivaSDK
 import AVFoundation
+import THEOplayerSDK
 
-struct AppEventConvivaReporter: AppEventProcessor {
+class AppEventConvivaReporter: AppEventProcessor {
+
+    
     let analytics: CISAnalytics
     let videoAnalytics: CISVideoAnalytics
     let adAnalytics: CISAdAnalytics
     let storage: ConvivaConnectorStorage
-
+    
+    init(analytics: CISAnalytics, videoAnalytics: CISVideoAnalytics, adAnalytics: CISAdAnalytics, storage: ConvivaConnectorStorage) {
+        self.analytics = analytics
+        self.videoAnalytics = videoAnalytics
+        self.adAnalytics = adAnalytics
+        self.storage = storage
+    }
+    
+    // This is a workaround for THEOSD-14780
+    // Access log entries for preroll ads come in before the player.ads.playing is updated, which means
+    // bitrate information gets reported to the main content instead, which is incorrect.
+    // Ads get loaded always before access log entries come, which we can use to report
+    // the ad bitrate to the correct endpoint.
+    private var prerollAdPlaying: Bool = false
+    
+    func adDidLoad(event: THEOplayerSDK.AdLoadedEvent)  {
+        prerollAdPlaying = true
```

Comment:
This will now set `prerollAdPlaying` to true for all ads, which can lead to false reporting in case an ad is loaded (which usually for mid-rolls happens a few segments in advance ~2-5 seconds on average) and then we receive an access log entry during main content playback.

Also, the flag name indicates that a preroll ad is playing, but it becomes true even if it's a midroll or postroll. So let's make sure we only enable this flag if it really is a preroll.

As for the fix itself, this solution is good and I validated that it reports now to the correct endpoint.